Userland/Libraries/LibSoftGPU/Clipper.h

```diff
@@ -44,15 +42,57 @@ class Clipper final {
     };
 
 public:
-    Clipper() { }
+    Clipper() = default;
 
     void clip_triangle_against_frustum(Vector<Vertex>& input_vecs);
 
 private:
-    bool point_within_clip_plane(const FloatVector4& vertex, ClipPlane plane);
-    Vertex clip_intersection_point(const Vertex& vec, const Vertex& prev_vec, ClipPlane plane_index);
+    Vertex clip_intersection_point(const Vertex& vec, const Vertex& prev_vec, ClipPlane plane);
+
+    template<ClipPlane plane>
+    constexpr bool point_within_clip_plane(const FloatVector4& vertex)
+    {
+        if constexpr (plane == ClipPlane::LEFT) {
+            return vertex.x() >= -vertex.w();
+        } else if constexpr (plane == ClipPlane::RIGHT) {
+            return vertex.x() <= vertex.w();
+        } else if constexpr (plane == ClipPlane::TOP) {
+            return vertex.y() <= vertex.w();
+        } else if constexpr (plane == ClipPlane::BOTTOM) {
+            return vertex.y() >= -vertex.w();
+        } else if constexpr (plane == ClipPlane::NEAR) {
+            return vertex.z() >= -vertex.w();
+        } else if constexpr (plane == ClipPlane::FAR) {
+            return vertex.z() <= vertex.w();
+        }
+
+        return false;
```

Comment:
Why is this not an error case?

Reply:
Probably should be, but I wasn't changing the semantics of the function. Will change.